src/Events/Admin/Notice/Rest_Api.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Shows an admin notice when our REST API endpoints are not available.
+ *
+ * @since
+ *
+ * @package TEC\Events\Admin\Notice
+ */
+
+namespace TEC\Events\Admin\Notice;
+
+use Tribe\Events\Views\V2\Rest_Endpoint as V2;
+
+/**
+ * Class Rest_Api
+ *
+ * Shows an admin notice when our REST API endpoints are not available.
+ *
+ * @since
+ */
+class Rest_Api {
+
+	/**
+	 * Notice Slug on the user options
+	 *
+	 * @since
+	 *
+	 * @var string
+	 */
+	private $slug = 'events-rest-api-notice';
+
+	/**
+	 * Blocked endpoint.
+	 *
+	 * @since
+	 *
+	 * @var string
+	 */
+	private $blocked_endpoint = '';
+
+	/**
+	 * Constructor.
+	 *
+	 * @since
+	 *
+	 * @return void
+	 */
+	public function hook(): void {
+		$slug = $this->slug;
+
+		tribe_notice(
+			$slug,
+			[ $this, 'notice' ],
+			[
+				'type'    => 'error',
+				'dismiss' => 1,
+				'wrap'    => 'p',
+			],
+			[ $this, 'should_display' ]
+		);
+	}
+
+	/**
+	 * Checks if we are in a TEC page or in the main Dashboard.
+	 *
+	 * @since
+	 *
+	 * @return boolean
+	 */
+	public function should_display(): bool {
+		global $pagenow;
+
+		if ( tribe( 'admin.helpers' )->is_screen() || 'index.php' === $pagenow ) {
+			return $this->is_rest_api_blocked() ? true : false;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checks if our endpoints are accessible.
+	 *
+	 * @since
+	 *
+	 * @return boolean
+	 */
+	public function is_rest_api_blocked(): bool {
+
+		$v1_api    = new \Tribe__Events__REST__V1__Main();
+		$event_api = get_rest_url( null, $v1_api->get_namespace() . '/events/' . $v1_api->get_version() );
```

Comment:
Lets get this from `Tribe__Events__REST__V1__Main::get_events_route_namespace`, turn that method `public` I see no reason to keep it `protected`, what do you think?